### PR TITLE
#181 Add games component

### PR DESCRIPTION
**Pull request by emptycoder**

### CODE REVIEW

File: Cooper.DAO/GameDAO.cs [hunk 1]
```diff
@@ -34,13 +34,22 @@ public GameDAO(IConfigProvider configProvider)
 
         #region Get methods
 
-        public GameDb Get(object id)
+        public GameDb Get(object obj)
         {
+            string name = obj.ToString();
```
Comment (Hunroll) [inline Cooper.DAO/GameDAO.cs]: ?
if obj is some class like array - you will have name="String[]"
Why can't we make a condition `if (obj is string name) { //find by name} else if (obj is long id) {...} else {some error}`

File: Cooper.Repositories/GameRepository.cs [hunk 2]
```diff
@@ -1,74 +1,74 @@
...
+
+            gameDAO.Update(gameDb);
+        }
+
+        public void Delete(long id)
+        {
+            gameDAO.Delete(id);
+        }
+
+        private Game ToGame(GameDb game)
```
Comment (Hunroll) [inline Cooper.Repositories/GameRepository.cs]: probably not needed and mapper should be called from methods

**Review by Hunroll — CHANGES_REQUESTED** [review]

File: Cooper/ClientApp/src.v2/app/app.module.ts [hunk 3]
```diff
@@ -16,7 +16,8 @@ import { CooperInterceptor } from 'src/assets/cooper.interceptor';
 import { GrowlModule } from 'primeng/primeng';
 import {MatIconModule} from '@angular/material/icon';
 import {MatProgressBarModule} from '@angular/material/progress-bar';
-import {AuthGuard} from '@guards';
+import { AuthGuard } from '@guards';
+import { SafePipe } from './pipes/safe.pipe';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/app.module.ts]: @pipes

File: Cooper/ClientApp/src.v2/app/components/avatar-card/avatar-card.component.scss [hunk 4]
```diff
@@ -1,10 +1,9 @@
 .avatar-card {   
     object-fit: cover;
     width: 230px;
-    height: auto;
+    height: 280px;
```
Comment (AndrePu) [inline Cooper/ClientApp/src.v2/app/components/avatar-card/avatar-card.component.scss]: Refactoring that file is not related to issue. It overlaps the changes that were made in #174 (Nickname bug on profile) pull-request

File: Cooper/ClientApp/src.v2/app/components/games-list/games-list.component.css [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .games-card {   
     object-fit: cover;
     width: 100%;
-    height: 292px;
+    height: 280px;
```
Comment (AndrePu) [inline Cooper/ClientApp/src.v2/app/components/games-list/games-list.component.css]: Refactoring that file is not related to issue. It overlaps the changes that were made in #174 (Nickname bug on profile) pull-request

File: Cooper/ClientApp/src.v2/app/components/user-connections-list/user-connections-list.component.css [hunk 6]
```diff
@@ -1,7 +1,7 @@
   .user-interaction-card {   
     object-fit: cover;
     width: 100%;
-    height: 292px;
+    height: 280px;
```
Comment (AndrePu) [inline Cooper/ClientApp/src.v2/app/components/user-connections-list/user-connections-list.component.css]: Refactoring that file is not related to issue. It overlaps the changes that were made in #174 (Nickname bug on profile) pull-request

File: Cooper/ClientApp/src.v2/app/layouts/game/game.layout.css [hunk 7]
```diff
@@ -0,0 +1,9 @@
+h1 {
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/game/game.layout.css]: .game-lauyout h1

File: Cooper/ClientApp/src.v2/app/components/avatar-card/avatar-card.component.scss [hunk 8]
```diff
@@ -4,10 +4,9 @@
     height: auto;
     margin-top: 30px;
     margin-bottom: 10px;
-    word-wrap: break-word;
```
Comment (AndrePu) [inline Cooper/ClientApp/src.v2/app/components/avatar-card/avatar-card.component.scss]: This line also needs to be restored

File: Cooper.Controllers/GameController.cs [hunk 9]
```diff
@@ -37,24 +37,31 @@ public IActionResult GetAll()
...
-            Game game = gameRepository.Get(id);
-
+            Game game = null;
+            if (long.TryParse(name, out long id))
+            {
+                game = gameRepository.Get(id);
+            }
+            else
+            {
+                game = gameRepository.Get(name);
```
Comment (horba) [inline Cooper.Controllers/GameController.cs:58]: is Name property in DB uniq?

File: Cooper/ClientApp/src.v2/app/layouts/top-panel/top-panel.layout.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import { Component } from '@angular/core';
-import { CoopNavBarItem } from 'src/app/models/coop-navbar-item';
+import { CoopNavBarItem } from 'src.v2/app/models/coop-navbar-item';
```
Comment (horba) [inline Cooper/ClientApp/src.v2/app/layouts/top-panel/top-panel.layout.ts]: @models

File: Cooper.Controllers/GameController.cs (hunk 9 above)
Comment (emptycoder) [reply Cooper.Controllers/GameController.cs:58]: Yes

**Review by horba — APPROVED** [review]

File: Cooper/ClientApp/src.v2/app/layouts/game/game.layout.ts [hunk 11]
```diff
@@ -1,10 +1,37 @@
...
   templateUrl: './game.layout.html',
   styleUrls: ['./game.layout.css']
 })
-export class GameLayoutComponent {
+export class GameLayoutComponent implements OnInit {
+  public game: Game;
+  public url: string;
 
+  constructor(private route: ActivatedRoute, private gameService: GamesService) {
+    this.game = {
```
Comment (AleksMorozova) [inline Cooper/ClientApp/src.v2/app/layouts/game/game.layout.ts]: I think it will be better to provide common progress indicator (In future). For now I suggest to remove this and use simple check in template whether current game exist [ use *ngIf].

**Review by AleksMorozova — CHANGES_REQUESTED** [review]

File: Cooper.Controllers/GameController.cs (hunk 9 above)
Comment (Hunroll) [reply Cooper.Controllers/GameController.cs:58]: I don't like... What if we have a game with name that will be parsable to long?
"2048", "10e2", "666", any other strange name) can we split searching by name and by id?

**Review by Hunroll — CHANGES_REQUESTED** [review]

**Review by Hunroll — APPROVED** [review]

**Review by Hunroll — APPROVED** [review]
